agent/internal/client/interface.go

```diff
@@ -140,6 +140,7 @@ type LoggerConfig struct {
 	Task               []LogOpts
 	SendToGlobalSender bool
 	AWSCredentials     *credentials.Credentials
+	ProjectVars        map[string]string
```

Comment:
Should we also pipe through these [global expansions](https://github.com/evergreen-ci/evergreen/blob/main/agent/command/expansion_write.go#L17-L23) here?

Reply:
There is also `${__project_aws_ssh_key_value}` in the [list of default expansions](https://docs.devprod.prod.corp.mongodb.com/evergreen/Project-Configuration/Project-Configuration-Files#default-expansions).

Reply:
Ah, so I know we talked about this yesterday. I should have left a comment explaining my rationale.

Of those expansions, only two (`GlobalGitHubTokenExpansion` and `GithubAppToken`) are populated as default expansions ([here](https://github.com/evergreen-ci/evergreen/blob/ca7e5d6fa44354c8fcf9b878545f3b8dab972aa9/model/project.go#L1133-L1134)). The others are populated by the ec2.assume_role command ([here](https://github.com/evergreen-ci/evergreen/blob/4dcce0dda1269f98a6fbde97aed31d12666e4b37/agent/command/assume_ec2_role.go#L123-L125)).

With regard to the github tokens, I think they're protected from being leaked in logs because they're not exposed as expansions to `shell.exec` or `subprocess.exec` ([here](https://github.com/evergreen-ci/evergreen/blob/66b48dce1b98cbc6d0dee8ce4c8523ea7f5e8a96/agent/command/exec.go#L184-L188)).

With regard to the AWS creds, they are temporary creds that expire after a few minutes, so it seemed okay to leave them. Though we could redact them too if we wanted to.

Reply:
Ah, I didn't read your whole comment.
If you feel like we shouldn't leak them I'll redact them.

Reply:
>Of those expansions, only two (GlobalGitHubTokenExpansion and GithubAppToken) are populated as default expansions ([here](https://github.com/evergreen-ci/evergreen/blob/ca7e5d6fa44354c8fcf9b878545f3b8dab972aa9/model/project.go#L1133-L1134)). The others are populated by the ec2.assume_role command ([here](https://github.com/evergreen-ci/evergreen/blob/4dcce0dda1269f98a6fbde97aed31d12666e4b37/agent/command/assume_ec2_role.go#L123-L125)).

The github token expansions are only protected from getting set as env vars, but they can still get expanded in the shell script [here](https://github.com/evergreen-ci/evergreen/blob/main/agent/command/shell.go#L253-L254).

We will have to figure out how to add the AWS expansions to the string replacer since they are set dynamically, that can be in follow up work.

Reply:
These are good points.
I've changed the strategy a bit and I think it should now cover all these.

afaict `__project_aws_ssh_key_value` is a project variable so I didn't include it.